### ISSUE TITLE
FIWARE/FB login problem

### ISSUE THREAD

**Issue by vladanminic** [pixelpark/ppnet#38]
Dear pixelpark team,

I just installed Social Network on VM Instance.
I'm having problems on login using my FIWARE or Facebook account. 

The same thing is happens here as well http://curious-electric.com/lab/ppnet/#/login 
(official demo page given in https://github.com/pixelpark/ppnet/blob/master/README.md) 

Can you please help me?

As a final result, I would like to login in my SocialNetwork using KeyRock IDM (installed on the same VM) but I don't have a clue how to that because of lack of information. Can you please post any link with helpful info?

Thank you in advance!

Best Regards,
Vladan

**Comment by dirkk0** [comment]
(cross posted (sort of) from the FIC2 page)
Hi Vladan,

you are right, I forgot to update the FB app.

To use (any) OAuth2 provider, one needs to register the app with the provider.
To do this in FB, you can 
a) follow these instructions: https://developers.facebook.com/docs/apps/register and
b) change this line https://goo.gl/RHdzuD with the magic number.
Then the FB login should work

The same is true for Keyrock.

So in a nutshell: register app in Keyrock and get the number and enter it in the file mentioned above.

Does this work for you?

Best,
Dirk

**Comment by vladanminic** [comment]
Hi Dirk,

it actually loads variables from this file (I'm running it from server folder [with ppnet proxy]: node server.js):
https://github.com/pixelpark/ppnet/blob/bf3e04d56bcb78a7381f26848340069f212d0c17/www/scripts/controller-1e44bc36.js#l43

I registered app on https://account.lab.fiware.org. I tried with: Client ID, Client Secret and with FiWare ID (I restarted server every time in order to accept changes, and I re-checked URL on Fiware login button in PPnet) but without luck.

Can you please point me to any link which describe how to do it properly?

Thank you very much!

Best,
Vladan

**Comment by vladanminic** [comment]
The problem was in auth URL in those files

app/vendor/hellojs/extended_hello.js:        auth: 'https://account.lab.fi-ware.org/authorize',
www/vendor/hellojs/extended_hello.js:        auth: 'https://account.lab.fi-ware.org/authorize',
www/scripts/vendor-07c80755.js:        auth: 'https://account.lab.fi-ware.org/authorize'

they should be auth: 'https://account.lab.fi-ware.org/oauth2/authorize'

**Comment by dirkk0** [comment]
Seems like you are very close.

First of all - thanks for pointing this out! It seems they changed a lot with Keyrock.

The 'mismatching_redirect_uri' seems to indicate that Keyrock can't find your server.  What did you enter as the 'Callback URL' in the application registration page? I think it needs to be pointed to your server.

**Comment by vladanminic** [comment]
Well, I tried with
http://myHostName:8500/#/login
http://myHostName:8500/#/
http://myHostName:8500/

but without success... Of course myHostName is just replacement for my address and I'm running it on port 8500 (Horizon is running on 8000)

Any suggestion? Tnx!

**Comment by vladanminic** [comment]
generated URL shoud be like this
https://account.lab.fiware.org/oauth2/authorize/?client_id=1b881b827**********5f1787970e44f&response_type=token&redirect_uri=http://myHostName:8500/index.html

but it generate it like this inistead:
https://account.lab.fiware.org/oauth2/authorize/?client_id=1b881b827**********5f1787970e44f&response_type=token&redirect_uri=http://myHostName:8500/index.html&display=popup&scope=&state={"client_id":"1b881b827**********5f1787970e44f","network":"fiware","display":"popup","callback":"_hellojs_aokob256","state":"","oauth_proxy":"https://auth-server.herokuapp.com/proxy","scope":["basic"],"oauth":{"version":2,"auth":"https://account.lab.fiware.org/oauth2/authorize","grant":"https://account.lab.fiware.org/oauth2/access_token"}}

Can you please help me to solve this?

**Comment by dirkk0** [comment]
ok, we need to take this to another level.
Could you consider a skype or hangout call next week, so that we might look into this in more detail?

**Comment by vladanminic** [comment]
Yes, of course. Please add me to your Skype contact list, my nick name is vladan.minic

**Comment by dirkk0** [comment]
Most likely a Keyrock problem ( see Vladans entry here: https://github.com/ging/fi-ware-idm/issues/8 ).
